### PR TITLE
[4.0] [RFC] Custom select Icons instead of SVGs

### PR DESCRIPTION
**Pull request by astridx**
Draft as basis for a discussion because of https://github.com/joomla/joomla-cms/pull/28680#issuecomment-615839018 and https://github.com/joomla/joomla-cms/pull/28755#issuecomment-618878957 .
(alternative for https://github.com/joomla/joomla-cms/pull/28755 and https://github.com/joomla/joomla-cms/pull/28680).
Also see:https://github.com/joomla/joomla-cms/issues/28673

### Summary of Changes
I implemented an idea of how we can use an Icon Font instead of SVGs

### Testing Instructions
At the moment this is a RFC.

Basically I prefer SVGs. But I think that the maintenance in an open source project with an icon font is easier.

### CODE REVIEW

File: administrator/templates/atum/scss/vendor/choicesjs/choices.scss [hunk 1]
```diff
@@ -33,6 +33,37 @@
...
+    background-color: var(--atum-bg-light);
+    color: var(--atum-text-dark);
+    content: "\107";
+    font-family: "Font Awesome 5 Free";
+    padding: $custom-select-padding;
+    position: absolute;
+    pointer-events: none;
+    top: 1px;
+
+    .moduleposition & {
```
Comment (astridx) [inline administrator/templates/atum/scss/vendor/choicesjs/choices.scss:48]: Here is an example of how you can change `color`, `background-colo`r and `icon`:

File: administrator/components/com_modules/layouts/joomla/form/field/modulespositionedit.php [hunk 2]
```diff
@@ -50,7 +50,7 @@
  */
 
 $attributes = array(
-	'class="' . $class . '"',
+	'class="moduleposition ' . $class . '"',
```
Comment (astridx) [inline administrator/components/com_modules/layouts/joomla/form/field/modulespositionedit.php:53]: Here is an example of how you can change color, background-color and icon

**Comment by coolcat-creations** [comment]
As far as I remember we tried to get rid of icon fonts in favour of svg ... not sure

**Comment by C-Lodder** [comment]
So all `<select>` elements need to be wrapped within a parent `<div>` to have a custom icon?

**Comment by infograf768** [comment]
> As far as I remember we tried to get rid of icon fonts in favour of svg ... not sure

svg are a pity to cope with, specially as background as we can't modify the color except if we create other svgs.
Icons are much much easier to deal with for devs/ bug fix

**Comment by Quy** [comment]
What about the frontend? It should not be dependent of fontawesome.

**Comment by astridx** [comment]
I just resolved a conflict in this draft. I became aware of the topic again. I think we just need a decision here. Otherwise, a contributor will do something that someone else will undo because he likes something else ...


These solutions are in my opinion technically better than this draft:
https://github.com/joomla/joomla-cms/pull/28262
https://github.com/joomla/joomla-cms/pull/28351


I made this draft because I think the solution in the draft is easier. Sometimes this is also advantageous (https://github.com/joomla/joomla-cms/pull/28787#issuecomment-619393718).


> @coolcat-creations 
 As far as I remember we tried to get rid of icon fonts in favour of svg ... not sure

Thank you for your feedback. You sure know more than I do. I wonder why you yourself continue to work with icon fonts instead of proposing a solution with SVGs: https://github.com/joomla/joomla-cms/pull/29016/files

> @Quy What about the frontend? It should not be dependent of fontawesome.

You're right. Thank you Quy. But this draft is only about Atum, i.e. the backend.


> @C-Lodder  So all `<select>` elements need to be wrapped within a parent `<div>` to have a custom icon?

Yes. But in the case of https://github.com/jshjohnson/Choices, an already existing div is used.

**Comment by Quy** [comment]
> But this draft is only about Atum, i.e. the backend.

But why do it differently on the backend and frontend for the same thing?

**Comment by astridx** [comment]
@Quy Please see:
https://github.com/joomla/joomla-cms/pull/28262#issuecomment-596442268
https://github.com/joomla/40-backend-template/pull/441#issuecomment-400840436

**Comment by astridx** [comment]
My solution is not a good one. I'm closing this here. https://github.com/joomla/joomla-cms/issues/28673 is not solved and should be reopened.
